flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/ExecNodeContext.java

```diff
@@ -176,4 +187,46 @@ public String toString() {
         }
         return new ExecNodeContext(metadata.name(), metadata.version());
     }
+
+    /**
+     * Create a configuration for the {@link ExecNode}, ready to be persisted to a JSON plan.
+     *
+     * @param execNodeClass The {@link ExecNode} class.
+     * @param plannerConfig The planner configuration (include the {@link TableConfig}).
+     * @return The {@link ExecNode} configuration, which contains the consumed options for the node,
+     *     defined by {@link ExecNodeMetadata#consumedOptions()}, along with their values.
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    public static <T extends ExecNode<?>> ReadableConfig newPersistedConfig(
+            Class<T> execNodeClass, ReadableConfig plannerConfig) {
+        Map<String, ConfigOption<?>> configOptions =
+                Stream.concat(
+                                ExecNodeMetadataUtil.TABLE_CONFIG_OPTIONS.stream(),
+                                ExecNodeMetadataUtil.EXECUTION_CONFIG_OPTIONS.stream())
+                        .collect(Collectors.toMap(ConfigOption::key, o -> o));
```

Comment:
we should also consider deprecated and fallback keys here. also add a test for it. e.g. an ExecNode with two versions. version 1 holds has a deprecated key, version 2 the new key, but both actually describe the same option.